xonsh/lexer.py

```diff
@@ -411,7 +429,7 @@ class Lexer:
 
     _tokens: tp.Optional[tp.Tuple[str, ...]] = None
 
-    def __init__(self, tolerant=False):
+    def __init__(self, tolerant=False, pymode=True):
         """
```

Comment:
is it possible to handle this in the parser? tokenizer can simply process it as numbers and operators in conjuction

Reply:
That was the approach which I thought about at the start, but there was a major problem which I didn't know how to solve: operator precedence.

If I had an expression like `x+2>0`, the `2>` would be grouped together by the tokenizer. But `x+2` needs to be evaluated first, so the parser would somehow need to know how to extract the `2` from the `IOREDIRECT` token and put it into an `Add` operation with the `x`. I couldn't think of a way to do this.

I thought that it would be easier to fix the `IOREDIRECT` tokens in the lexer so that I wouldn't have to change any of the parser's grammar.

Reply:
Good then